packages/frontend/src/components/Routing.js

```diff
@@ -224,21 +229,43 @@ class Routing extends Component {
     }
 
     startPollingTokenFiatValue = () => {
-        const { fetchTokenFiatValues } = this.props;
+        const { fetchTokenFiatValues, fetchTokenUSDNFiatValues } = this.props;
 
         const handlePollTokenFiatValue = async () => {
-            await fetchTokenFiatValues().catch(() => { });
+            await fetchTokenFiatValues().catch(() => {});
             if (this.pollTokenFiatValue) {
-                this.pollTokenFiatValue = setTimeout(() => handlePollTokenFiatValue(), 30000);
+                this.pollTokenFiatValue = setTimeout(
+                    () => handlePollTokenFiatValue(),
+                    30000
+                );
             }
         };
-        this.pollTokenFiatValue = setTimeout(() => handlePollTokenFiatValue(), 30000);
-    }
+        this.pollTokenFiatValue = setTimeout(
+            () => handlePollTokenFiatValue(),
+            30000
+        );
+
+        const handlePollTokenFiatValueUSDN = async () => {
```

Comment:
If I understand correctly, you suggest combining calls to get fit values ​​into one thunk and transfer pulling to the FungibleTokens component in Wallet.js, and call it only once in routing?

Reply:
Yes that seems right. The main point here though is to generalize `fetchTokenFiatValues` to fetch multiple tokens rather than adding new thunks per token. I also suspect that we will be augmenting the `/fiat` endpoint to take input but still confirming if this will make it in time for this PR.